### PR TITLE
Home work 4

### PR DESCRIPTION
**Pull request by olhaterefenko**

### CODE REVIEW

File: terefenko/HW4/Task_4_1.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+"""
+
+
+years = int(input("enter year:"))
+century = years // 100 + 1
+print("century is:", century)
+
+if years % 4 == 0 and years / 100 != 0:
+    print("years is intercalary")
+elif years % 400 == 0:
```
Comment (katemalash) [inline terefenko/HW4/Task_4_1.py:16]: Is it possible to place all true conditions in one branch?

File: terefenko/HW4/Task_4_2.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    side_a = float(input('side_a='))
+    side_b = float(input('side_b='))
+    square_rectangle = side_a * side_b
+    print('Square rectangle:', square_rectangle)
+elif figure == 'triangle':
+    print('The lengths of the triangles sides is:')
+    side_a = float(input('side_a='))
+    side_b = float(input('side_b='))
+    side_c = float(input('side_c='))
+    perimeter = (side_a + side_b + side_c)/2
```
Comment (katemalash) [inline terefenko/HW4/Task_4_2.py:23]: Is it correct to name this variable perimeter?

File: terefenko/HW4/Task_4_2.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    square_triangle = ((perimeter-side_a)*(perimeter-side_b)*(perimeter-side_c)) ** 0.5
+    print('Square triangle:', square_triangle)
+elif figure == 'circle':
+    print('The lengths radius of the circle is:')
+    radius_circle = float(input('radius_circle ='))
+    square_circle = 2 * 3.14 * radius_circle
```
Comment (katemalash) [inline terefenko/HW4/Task_4_2.py:29]: Are you sure this is correct formula?

File: terefenko/HW4/Task_4_3.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+if number < 0:
+    print("the number is incorrect")
+elif 0 <= number <= 9:
+    print("positive one-digit")
+elif 10 <= number <= 99:
+    print("negative two- digit")
+elif 100 <= number <= 999:
+    print("negative tree-digit")
+else:
+    print("the number is another")
```
Comment (katemalash) [inline terefenko/HW4/Task_4_3.py:21]: Is it possible to count the length of input value?

File: terefenko/HW4/Task_4_4.py [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+        print("There is no stage")
+    elif side_hall <= diameter_stage | side_hall <= 0:
+        # Enter side of hall , our hall is square.
+        print("There is no passage!")
+    elif side_hall >= radius_stage:
+        print("Passage:", k_passage)
+    else:
+        print("You are wrong !")
+else:
+    print("Something went wrong!")
```
Comment (katemalash) [inline terefenko/HW4/Task_4_4.py:30]: It works

File: terefenko/HW4/Task_4_5.py [hunk 6]
```diff
@@ -0,0 +1,43 @@
+"""
+@author: Olha Terefenko.
+date: 2.08.21.
+condition:
+The amount of money is know . Exchange it with 200, 100, 10 banknotes and 1UAN coin , if possible.
+"""
+import re
+
+regex_pattern = '^[0-9]+$'
```
Comment (katemalash) [inline terefenko/HW4/Task_4_5.py:9]: Maybe .isdigit() is more effective way?)

File: terefenko/HW4/Task_4_5.py [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+    if amount_money >= 10:
+        counter_10 = amount_money // nominal_10
+        amount_money = amount_money - counter_10 * nominal_10
+        print("Banknote of nominal {0}: {1}".format(nominal_10, counter_10))
+    if amount_money >= 1:
+        counter_1 = amount_money // nominal_1
+        amount_money = amount_money - counter_1 * nominal_1
+        print("Banknote of nominal {0}: {1}".format(nominal_1, counter_1))
+else:
+    print("You input incorrect value. Only digit is allowed.")
```
Comment (katemalash) [inline terefenko/HW4/Task_4_5.py:43]: Good decision

File: terefenko/HW4/Task_4_6.py [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+        if number_place <= 36:
+            print("Place of carriage in compartment!")
+        else:
+            print("Place of carriage in side!")
+        if number_place % 2 == 0:
+            print("Place of carriage is top!")
+        else:
+            print("Place of carriage is bottom!")
+else:
+    print("The place number is incorrect!")
```
Comment (katemalash) [inline terefenko/HW4/Task_4_6.py:27]: Nice!

**Review by katemalash — CHANGES_REQUESTED** [review]
Please fix calculations for circle square

File: terefenko/HW4/Task_4_3.py (hunk 4 above)
Comment (olhaterefenko) [reply terefenko/HW4/Task_4_3.py:21]: Yes, I'll correct it.

File: terefenko/HW4/Task_4_5.py (hunk 6 above)
Comment (olhaterefenko) [reply terefenko/HW4/Task_4_5.py:9]: Maybe)

File: terefenko/HW4/Task_4_1.py (hunk 1 above)
Comment (olhaterefenko) [reply terefenko/HW4/Task_4_1.py]: done

File: terefenko/HW4/Task_4_2.py (hunk 2 above)
Comment (olhaterefenko) [reply terefenko/HW4/Task_4_2.py]: fixed

File: terefenko/HW4/Task_4_2.py (hunk 3 above)
Comment (olhaterefenko) [reply terefenko/HW4/Task_4_2.py]: fixed

File: terefenko/HW4/Task_4_3.py (hunk 4 above)
Comment (olhaterefenko) [reply terefenko/HW4/Task_4_3.py]: done

**Review by katemalash — APPROVED** [review]
Well Now!
Please merge this pull request

**Pull request merged**